python/ray/data/preprocessor.py

```diff
@@ -8,6 +8,7 @@
 
 if TYPE_CHECKING:
     import pandas as pd
+    import numpy as np
```

Comment:
Small nit: NumPy is available as a Ray dependency so we don't need to guard it under `TYPE_CHECKING`.